week05/clause_test.go

```diff
@@ -0,0 +1,76 @@
+package demo
+
+import (
+	"testing"
+)
+
+func Test_clause_String(t *testing.T) {
+	t.Parallel()
+
+	table := []struct {
+		name string
+		cls  Clauses
+		exp  string
+	}{
+		{
+			name: "matching clauses",
+			cls:  Clauses{"name": "Sir Canon Doyle", "job": "Investigator"},
+			exp:  `"job" = "Investigator" and "name" = "Sir Canon Doyle"`},
+		{
+			name: "unmatching clauses",
+			cls:  Clauses{"spiderman": "peter parker", "batman": "bruce wayne"},
+			exp:  `"batman" = "bruce wayne" and "spiderman" = "peter parker"`},
+		{
+			name: "empty clauses",
+			cls:  Clauses{},
+			exp:  ``},
+	}
+
+	for _, tt := range table {
+		t.Run(tt.name, func(t *testing.T) {
+			act := tt.cls.String()
+
+			if act != tt.exp {
+				t.Fatalf("%s expected %q and %q", tt.name, tt.exp, act)
+			}
+		})
+	}
+}
+
+func Test_clause_Match(t *testing.T) {
+	t.Parallel()
+
+	table := []struct {
+		name string
+		cls  Clauses
+		mdl  Model
+		exp  bool
+	}{
+		{
+			name: "true case",
+			cls:  Clauses{"name": "Sir Canon Doyle", "job": "Investigator"},
+			mdl:  Model{"name": "Sir Canon Doyle", "job": "Investigator"},
+			exp:  true},
```

Comment:
Put the closing curly brace on a new line to match the opening one. I found this very confusing. At first I couldn't figure out how the code was even compiling.

```suggestion
		{
			name: "true case",
			cls:  Clauses{"name": "Sir Canon Doyle", "job": "Investigator"},
			mdl:  Model{"name": "Sir Canon Doyle", "job": "Investigator"},
			exp:  true,
		},
```